neps/nep-0452.md

```diff
@@ -0,0 +1,234 @@
+- The contract will map the `pubKey1` to the desired balance for the linkdrop.
+- The contract will then add the `pubKey1` as a function call access key with the ability to call `claim` and `create_account_and_claim`. This means that anyone with the `privKey1` (see above), can sign a transaction on behalf of this contract (signer id set to contract id) with a function call to call one of the mentioned functions to claim the assets.
+
+#### Claiming a linkdrop without a NEAR Account
+
+A user with _no_ wallet that is claiming the assets associated with an existing public key, already registered on the linkdrop contract:
+
+- generates a new keypair (`pubKey2`, `privKey2`) locally. (This new keypair is not written to chain)
+- chooses a new account ID such as benji.near.
+- calls `create_account_and_claim`. The transaction is signed on behalf of the linkdrop contract (`signer_id` is set to the contract address) using `privKey1`.
+  - the args of this function call will contain both `pubKey2` (which will be used to create a full access key for the new account) and the account ID itself.
+  - the linkdrop contract will delete the access key associated with `pubKey1` so that it cannot be used again.
+  - the linkdrop contract will create the new account and transfer the funds to it alongside any other assets.
+- the user will be able to sign transactions on behalf of the new account using `privKey2`.
+
+#### Claiming a linkdrop with a NEAR Account
+
+A user with an _existing_ wallet that is claiming the assets with an existing public key, already registered on the linkdrop contract:
+
+- calls `claim`. The transaction is signed on behalf of the linkdrop contract (`signer_id` is set to the contract address) using `privKey1`.
+  - the args of this function call will simply contain the user's existing account ID.
+  - the linkdrop contract will delete the access key associated with `pubKey1` so that it cannot be used again.
+  - the linkdrop contract will transfer the funds to that account alongside any other assets.
+
+```ts
+/// Information about a specific public key.
+type KeyInfo = {
+   /// How much Gas should be attached when the key is used to call `claim` or `create_account_and_claim`.
+   /// It is up to the smart contract developer to calculate the required gas (which can be done either automatically on the contract or on the client-side).
+   required_gas: string,
+
+   /// yoctoNEAR$ amount that will be sent to the account that claims the linkdrop (either new or existing)
+   /// when the key is successfully used.
+   yoctoNEAR: string,
+
+   /// If using the NFT standard extension, a set of NFTData can be linked to the public key
+   /// indicating that all those assets will be sent to the account that claims the linkdrop (either new or
+   /// existing) when the key is successfully used.
+   nft_data: NFTData[] | null,
```

Comment:
Well, it's a general comment. If there will be another token type, then it will be the prefix which will change (eg `nft_`) rather than suffix (`_data`).

Would this change break any existing environment?

Reply:
I also don't see any value in `_data` suffix as well, and I would also prefer dropping it as I feel it is just cleaner to have:

```json
{
  "yoctoNEAR": "123",
  "nft_list": [{...}],
  "ft_list": [{...}, {...}],
}
```

"data" is too generic, and just reading "nft_data" does not hint that it is going to be more than a single NFT there. "list" at least communicates that there will be a list of them. "nfts" and "fts" are not great as it might become misleading if "NFTS" / "FTS" standard will appear.

BUT, I won't have hard feelings if we keep this useless "_data" suffix, so if you feel you like it or it would trigger too many changes, I am not going to reject the NEP because of it.